lib/base/tlsutility.hpp

```diff
@@ -69,7 +69,7 @@ String SHA256(const String& s);
 String RandomString(int length);
 String BinaryToHex(const unsigned char* data, size_t length);
 
-bool VerifyCertificate(const std::shared_ptr<X509>& caCertificate, const std::shared_ptr<X509>& certificate, const String& crlFile);
+bool VerifyCertificate(const std::shared_ptr<X509>& caCertificate, const std::shared_ptr<X509>& certificate, const String& crlFile, STACK_OF(X509) *chain = nullptr);
```

Comment:
Ok, as far as I see, this one now allows a whole chain...

Reply:
Actually `chain` is just a bundle of (untrusted) certificates which OpenSSL will use in its verification process to check if the supplied target certificate is valid according to its trusted ca-certificate.